packages/report-server/src/reportBuilder/transform/parser/tranformParser.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { ExpressionParser } from '@tupaia/expression-parser';
+import { Row, FieldValue } from '../../types';
+
+type RowLookup = {
+  [key: string]: FieldValue[];
+};
+
+/**
+ * Lookups object for rows within the transform data
+ *
+ * eg. if rows = [{BCD1: 5}, {BCD1: 8}, {BCD1: 4}], and currentRow = 1
+ * '$row.BCD1' => 8
+ * '$all.BCD1' => [5, 8, 4]
+ * '$allPrevious.BCD1' => [5, 8]
+ * '$where(f($otherRow) = $otherRow.BCD1 < $row.BCD1).BCD1' => [5, 4]
+ * '$row.BCD1 + sum($allPrevious.BCD1)' => 21
+ */
+type Lookups = {
```

Comment:
If anyone has a better name than `Lookup` I'm all ears. I'm not too happy with it but nothing better comes to mind...